coda/coda_mdstore/tests/test_utils.py

```diff
@@ -1,14 +1,573 @@
+
+    def test_name_not_set(self, bag_xml):
+        del bag_xml.name
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+
+        assert bag is None
+        assert bag_infos is None
+        assert error == "Unable to set 'name' attribute"
+
+    def test_name_is_set(self, bag_xml):
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+
+        assert bag.name == bag_xml.name
+        assert error is None
+
+    def test_fileCount_not_set(self, bag_xml):
+        del bag_xml.fileCount
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+
+        assert bag is None
+        assert bag_infos is None
+        assert error == "Unable to set 'files' attribute"
+
+    def test_fileCount_is_set(self, bag_xml):
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+
+        assert bag.files == str(bag_xml.fileCount)
+        assert error is None
+
+    def test_payloadSize_not_set(self, bag_xml):
+        del bag_xml.payloadSize
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+
+        assert bag is None
+        assert bag_infos is None
+        assert "Unable to set 'size' attribute" in error
+
+    def test_payloadSize_is_set(self, bag_xml):
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+
+        assert bag.size == bag_xml.payloadSize
+        assert error is None
+
+    def test_lastStatus_not_set(self, bag_xml):
+        del bag_xml.lastStatus
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+
+        assert bag.last_verified_status == 'pass'
+        assert error is None
+
+    def test_lastStatus_is_set(self, bag_xml):
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+
+        assert bag.last_verified_status == bag_xml.lastStatus
+        assert error is None
+
+    def test_lastVerified_not_set(self, bag_xml):
+        del bag_xml.lastVerified
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+
+        assert isinstance(bag.last_verified_date, datetime)
+        assert error is None
+
+    def test_lastVerified_is_set(self, bag_xml):
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+
+        assert isinstance(bag.last_verified_date, datetime)
+        assert error is None
+
+    def test_bagitVersion_not_set(self, bag_xml):
+        del bag_xml.bagitVersion
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+
+        assert bag.bagit_version == ''
+        assert error is None
+
+    def test_bagitVersion_is_set(self, bag_xml):
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+
+        assert bag.bagit_version == str(bag_xml.bagitVersion)
+        assert error is None
+
+    def test_baggingDate_not_set(self, bag_xml):
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+
+        assert isinstance(bag.bagging_date, datetime)
+        assert error is None
+
```

Comment:
No I left it out intentionally.

**Why, you ask?**
I have a personal preference to avoid testing the state of datetime objects. Perhaps, I should get over it.